core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2938,13 +2959,21 @@ private void scheduleAsyncPersistenceInternal(LockedInodePath inodePath) throws
   }
 
   @Override
-  public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFiles)
+  public FileSystemCommand workerHeartbeat(long workerId, List<Long> persistedFiles,
+      WorkerHeartbeatOptions options)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException,
       IOException {
-    for (long fileId : persistedFiles) {
+
+    List<String> persistedUfsFingerprints = options.getPersistedUfsFingerprintList();
+    boolean hasPersistedFingerprints = persistedUfsFingerprints.size() == persistedFiles.size();
```

Comment:
Is it ever possible that the sizes are non zero but also not equal?

Reply:
That "should" not happen, since on the worker side, we are constructing both lists from the map entries.